src/Service/ValidatorClassNameResolver.php

```diff
@@ -0,0 +1,47 @@
+<?php declare(strict_types = 1);
+
+namespace SaschaEgerer\PhpstanTypo3\Service;
+
+use PHPStan\Reflection\ReflectionProvider;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\Type;
+use ReflectionMethod;
+use TYPO3\CMS\Extbase\Validation\ValidatorResolver;
+
+final class ValidatorClassNameResolver
+{
+
+	/** @var ReflectionProvider */
+	private $reflectionProvider;
+
+	public function __construct(ReflectionProvider $reflectionProvider)
+	{
+		$this->reflectionProvider = $reflectionProvider;
+	}
+
+	public function resolve(Type $type): ?string
+	{
+		if ( ! $type instanceof ConstantStringType) {
+			return null;
+		}
+
+		if ($this->reflectionProvider->hasClass('TYPO3\CMS\Extbase\Validation\ValidatorClassNameResolver')) {
```

Comment:
```suggestion
		if ($this->reflectionProvider->hasClass(\TYPO3\CMS\Extbase\Validation\ValidatorClassNameResolver::class)) {
```

Reply:
It's still a string

Reply:
I know. I am using this because otherwise phpstan is complaining. In rector itself we are using always the String instead of ::class. I have changed it, but i had to ignore the arising error